ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/utils/KafkaUtil.scala

```diff
@@ -29,29 +29,40 @@ import scala.collection.mutable
 private[hyperdrive] object KafkaUtil {
   private val logger = LogManager.getLogger
 
-  def getAtLeastNLatestRecordsFromPartition[K, V](consumer: KafkaConsumer[K, V], topicPartition: TopicPartition, numberOfRecords: Int)
+  def getAtLeastNLatestRecordsFromPartition[K, V](consumer: KafkaConsumer[K, V], numberOfRecords: Map[TopicPartition, Long])
     (implicit kafkaConsumerTimeout: Duration): Seq[ConsumerRecord[K, V]] = {
-    consumer.assign(Seq(topicPartition).asJava)
-    val endOffsets = consumer.endOffsets(Seq(topicPartition).asJava).asScala
-    if (endOffsets.size != 1) {
-      throw new IllegalStateException(s"Expected exactly 1 end offset, got ${endOffsets}")
-    }
-    val partition = endOffsets.keys.head
-    val offset = endOffsets.values.head
+    consumer.assign(numberOfRecords.keySet.asJava)
+    val endOffsets = consumer.endOffsets(numberOfRecords.keySet.asJava).asScala
+    val topicPartitions = endOffsets.keySet
 
     var records: Seq[ConsumerRecord[K, V]] = Seq()
-    var offsetLowerBound = offset
-    while (records.size < numberOfRecords && offsetLowerBound != 0) {
-      offsetLowerBound = Math.max(0, offsetLowerBound - numberOfRecords)
-      consumer.seek(partition, offsetLowerBound)
-      records = getMessagesAtLeastToOffset(consumer, Map(topicPartition -> offset))
+    val offsetLowerBounds = mutable.Map(endOffsets.mapValues(_ + 0L).toSeq: _*)
```

Comment:
I am just curious, why do you need  + 0L ?

Reply:
There is no implicit conversion from `java.lang.Long` to `scala.Long` inside collections, but I should have used `Long2long`. Fixed it